src/qutip_qip/decompositions/single_decompositions.py

```diff
@@ -0,0 +1,226 @@
+    ----------
+    input_gate : :class:`qutip.Qobj`
+        The matrix that's supposed to be decomposed should be a Qobj.
+    """
+    angle_list = _angles_for_ZYZ(input_gate, single_qubit)
+    alpha = angle_list[0]
+    beta = angle_list[2]
+    theta = angle_list[1]
+    global_phase_angle=angle_list[3]
+
+    # for string in circuit diagram
+    alpha_string = alpha/np.pi
+    beta_string = beta/np.pi
+    theta_string = theta/np.pi
+    global_phase_angle_string = global_phase_angle/np.pi
+
+    Phase_gate = Gate("GLOBALPHASE",targets=[target], arg_value=global_phase_angle, arg_label=r'{:0.2f} \times \pi'.format(global_phase_angle_string))
+    Rz_alpha = Gate("RZ",targets=[target], arg_value=alpha, arg_label=r'{:0.2f} \times \pi'.format(alpha_string))
+    Ry_theta = Gate("RY",targets=[target], arg_value=theta, arg_label=r'{:0.2f} \times \pi'.format(theta_string))
+    Rz_beta = Gate("RZ",targets=[target], arg_value=beta, arg_label=r'{:0.2f} \times \pi'.format(beta_string))
+
+    return(Rz_alpha, Ry_theta, Rz_beta, Phase_gate)
+
+def _ZXZ_rotation(input_gate, num_of_qubits, target):
+    r""" An input 1-qubit gate is expressed as a product of rotation matrices
+    :math:`\textrm{R}_z` and :math:`\textrm{R}_x`.
+
+    Parameters
+    ----------
+    input_gate : :class:`qutip.Qobj`
+        The matrix that's supposed to be decomposed should be a Qobj.
+    """
+    angle_list = _angles_for_ZYZ(input_gate, single_qubit)
+    alpha = angle_list[0]
+    alpha = alpha - np.pi/2
+    beta = angle_list[2]
+    beta = beta + np.pi/2
+    theta = angle_list[1]
+    global_phase_angle=angle_list[3]
+
+    # for string in circuit diagram
+    alpha_string = alpha/np.pi
+    beta_string = beta/np.pi
+    theta_string = theta/np.pi
+    global_phase_angle_string = global_phase_angle/np.pi
+
+    Phase_gate = Gate("GLOBALPHASE",targets=[target], arg_value=global_phase_angle, arg_label=r'{:0.2f} \times \pi'.format(global_phase_angle_string))
+    Rz_alpha = Gate("RZ",targets=[target], arg_value=alpha, arg_label=r'{:0.2f} \times \pi'.format(alpha_string))
+    Rx_theta = Gate("RX",targets=[target], arg_value=theta, arg_label=r'{:0.2f} \times \pi'.format(theta_string))
+    Rz_beta = Gate("RZ",targets=[target], arg_value=beta, arg_label=r'{:0.2f} \times \pi'.format(beta_string))
+
+    return(Rz_alpha, Rx_theta, Rz_beta, Phase_gate)
+
+
+_rotation_matrices_dictionary ={"ZYZ": _ZYZ_rotation,
+                                "ZXZ": _ZXZ_rotation,
+                                }
+
+def decompose_to_rotation_matrices(input_gate, num_of_qubits, target, method):
+    r""" An input 1-qubit gate is expressed as a product of rotation matrices
+    :math:`\textrm{R}_i` and :math:`\textrm{R}_j`.
+
+    Here, :math:`i \neq j` and :math:`i, j \in {x, y, z}`
+
+    .. math::
+
+        U = \begin{bmatrix}
+            a       & b \\
+            -b^*       & a^*  \\
+            \end{bmatrix} = \textrm{R}_i(\alpha)  \textrm{R}_j(\theta)  \textrm{R}_i(\beta)
+
```

Comment:
> num_of_qubits is actually not needed here because it is by definition single-qubit gates

Yes, it is needed to check if a target is provided for larger number of qubits. Here's the check :

```python
try:
        assert num_of_qubits == 1
except AssertionError:
        if target is None and num_of_qubits >1:
            raise GateError("This method is valid for single qubit gates only. Provide a target qubit for single qubit gate.")
```